src/sprite_3.c

```diff
@@ -3,7 +3,25 @@
 #include "sprite.h"
 #include "flags.h"
 
-extern const u8 gUnknown_080984DC[][2];
+const u8 gOamShapesSizes[][2] = {
+    // Square
+    { 8, 8 },
+    { 16, 16 },
+    { 32, 32 },
+    { 64, 64 },
+
+    // Horizontal
+    { 16, 8 },
+    { 32, 8 },
+    { 32, 16 },
+    { 64, 32 },
+
+    // Vertical
+    { 8, 16 },
+    { 8, 32 },
+    { 16, 32 },
+    { 32, 64 },
+};
 
 void sub_80051E8(Sprite *sprite)
```

Comment:
So this is like `ApplyEffects`?

Reply:
That name's too general for a global procedure.
Also, I don't think "effects" is right? How did you infer that name?

Reply:
I'm not saying that should be the name. This functions is generally called with the sprite `RenderAnimationFrame` function call.

Maybe you are right that it's not effects. When I say effects I am referring to `unk10` which is like a flag of effects which apply to the sprite (though this could also be `flags`) when this function is called the `OamData` of the sprite is updated with the result of effects currently applied to the sprite? rotation etc?

Reply:
Is it not `Sprite_ApplyOamAttributes(...)`